nvflare/recipe/fedavg.py

```diff
@@ -0,0 +1,340 @@
+        framework: FrameworkType = FrameworkType.PYTORCH,
+        server_expected_format: ExchangeFormat = ExchangeFormat.NUMPY,
+        params_transfer_type: TransferType = TransferType.FULL,
+        model_persistor: Optional[ModelPersistor] = None,
+        custom_persistor: Optional[ModelPersistor] = None,
+    ):
+        # Validate inputs internally
+        v = _FedAvgValidator(
+            name=name,
+            initial_model=initial_model,
+            initial_params=initial_params,
+            min_clients=min_clients,
+            num_rounds=num_rounds,
+            train_script=train_script,
+            train_args=train_args,
+            aggregator=aggregator,
+            aggregator_data_kind=aggregator_data_kind,
+            launch_external_process=launch_external_process,
+            command=command,
+            framework=framework,
+            server_expected_format=server_expected_format,
+            params_transfer_type=params_transfer_type,
+            model_persistor=model_persistor,
+            custom_persistor=custom_persistor,
+        )
+
+        self.name = v.name
+        self.initial_model = v.initial_model
+        self.initial_params = v.initial_params
+        self.min_clients = v.min_clients
+        self.num_rounds = v.num_rounds
+        self.train_script = v.train_script
+        self.train_args = v.train_args
+        self.aggregator = v.aggregator
+        self.aggregator_data_kind = v.aggregator_data_kind
+        self.launch_external_process = v.launch_external_process
+        self.command = v.command
+        self.framework = v.framework
+        self.server_expected_format = v.server_expected_format
+        self.params_transfer_type = v.params_transfer_type
+        self.model_persistor = v.model_persistor
+        self.custom_persistor = v.custom_persistor
+
+        # Validate that only one of initial_model or initial_params is provided
+        if self.initial_model is not None and self.initial_params is not None:
+            raise ValueError(
+                "Cannot provide both initial_model and initial_params. "
+                "Use initial_model for PyTorch/TensorFlow, initial_params for sklearn."
+            )
+
+        # Validate RAW framework has custom_persistor
+        if self.framework == FrameworkType.RAW and self.custom_persistor is None:
+            raise ValueError(
+                "custom_persistor is required when framework=FrameworkType.RAW. "
+                "Use framework-specific wrappers (e.g., SklearnFedAvgRecipe) or provide a custom persistor."
+            )
+
+        # Create BaseFedJob - all frameworks use it for consistency
+        # Provide default TBAnalyticsReceiver for PT/TF only
+        analytics_receiver = None
+        if self.framework in (FrameworkType.PYTORCH, FrameworkType.TENSORFLOW):
+            from nvflare.app_opt.tracking.tb.tb_receiver import TBAnalyticsReceiver
+
+            analytics_receiver = TBAnalyticsReceiver()
```

Comment:
this is already done by child class I think

Reply:
you are right, in that case, let's push these to child classes, make this class general purpose